feathr-example/bureau_bureau-balance.ipynb

```diff
@@ -0,0 +1,1224 @@
+    "1. combinig/joining both datasets, which will be aggregated in line with primary key (\"SK_ID_CURR) of application_train (target dataframe) with the following features:    \n",
+    "    1. count aggregation features created\n",
+    "    1. average aggregation features created\n",
+    "    1. debt:credit ratio feature created\n",
+    "    1. rolling window credit loan status feature will be created and joined to parent dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# two datasource pointing to same csv, limitation that you could not mix\n",
+    "# pass through and aggregated features. By separating, it must have different datasource (datasource name)\n",
+    "\n",
+    "# source for pass through features\n",
+    "# \"TRAN_DATE\" column created on on the \"datasource_prepocessing\" method.\n",
+    "bureau_source_core = HdfsSource(name=\"bureauSourceCore\",\n",
+    "                          path=\"abfss://feathrhomecreditcafs@feathrhomecreditcasto.dfs.core.windows.net/home_credit_data/bureau.csv\",\n",
+    "                          preprocessing=bureau_preprocessing,\n",
+    "                          event_timestamp_column=\"TRAN_DATE\",\n",
+    "                          timestamp_format=\"yyyy-MM-dd HH:mm:ss\"\n",
+    "                          )\n",
+    "\n",
+    "# key definition for bureau datasource\n",
+    "key_SK_ID_BUREAU = TypedKey(key_column=\"SK_ID_BUREAU\",\n",
+    "                       key_column_type=ValueType.INT32,\n",
+    "                       description=\"SK ID Bureau\",\n",
+    "                       full_name=\"bureau.SK_ID_BUREAU\")\n",
+    "\n",
+    "key_SK_ID_CURR = TypedKey(key_column=\"SK_ID_CURR\",\n",
+    "                       key_column_type=ValueType.INT32,\n",
+    "                       description=\"SK ID CURR\",\n",
+    "                       full_name=\"bureau.SK_ID_CURR\")\n",
+    "\n",
+    "# pass through columns of BUREAU datasource CSV\n",
+    "f_SK_ID_CURR = Feature(name=\"f_SK_ID_CURR\",\n",
+    "                        key=key_SK_ID_BUREAU, \n",
+    "                        feature_type=INT32, \n",
+    "                        transform=\"SK_ID_CURR\")\n",
+    "\n",
+    "f_SK_ID_BUREAU  = Feature(name=\"f_SK_ID_BUREAU\",\n",
+    "                        key=key_SK_ID_BUREAU, \n",
+    "                        feature_type=STRING, \n",
+    "                        transform=\"SK_ID_BUREAU\")\n",
+    "\n",
+    "f_CREDIT_ACTIVE = Feature(name=\"f_CREDIT_ACTIVE\",\n",
+    "                        key=key_SK_ID_BUREAU, \n",
+    "                        feature_type=STRING, \n",
+    "                        transform=\"CREDIT_ACTIVE\")\n",
+    "\n",
+    "f_CREDIT_CURRENCY = Feature(name=\"f_CREDIT_CURRENCY\",\n",
+    "                        key=key_SK_ID_BUREAU, \n",
+    "                        feature_type=STRING, \n",
```

Comment:
Am I understanding this correctly? Are all feature just a passthrough? No transformations are defined here? If so, it might be worth mentioning so somewhere. Also, is it possible to move some of the transformations here? I'm thinking from the perspective of showcasing the functionality that Feathr provides. Otherwise the demo won't showcase a lot. Only, if it is possible and **where it makes sense.**

Reply:
Agree on this, though these anchor features are all passthrough, we can still show some more transformation abilities such as defining a window function like documented here https://github.com/feathr-ai/feathr#defining-window-aggregation-features-with-point-in-time-correctness

Reply:
The derived features placed here are just for testing - just to see how it works on feathr. The feast implementation doesn't have this, I would be removing this.

Reply:
Sure, I think it also makes sense to keep feathr example as close as the feathr. Maybe we can add more examples to feathr later on just like the streaming examples.